extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayRecorder.java

```diff
@@ -128,6 +128,10 @@ public void doStartActions(String dataSourceName) {
         if (flywayContainer.isCleanAtStart()) {
             flywayContainer.getFlyway().clean();
         }
+        if (flywayContainer.isCleanOnValidationError()
+                && !flywayContainer.getFlyway().validateWithResult().validationSuccessful) {
+            flywayContainer.getFlyway().clean();
+        }
         if (flywayContainer.isValidateAtStart()) {
             flywayContainer.getFlyway().validate();
         }
```

Comment:
Won't this result in _two_ validations being run?

Wouldn't this make more sense?

```suggestion
        if (flywayContainer.isValidateAtStart()) {
            var result = flywayContainer.getFlyway().validateWithResult();
            
            if (flywayContainer.isCleanOnValidationError() && !result.validationSuccessful) {
                flywayContainer.getFlyway().clean();
            }
        }
```

Reply:
Makes perfekt sense 👍

Reply:
There are test failures.I will take a Look.

Reply:
The problem here is that before the change validate-at-start threw a FlywayValidationException if the validation failed. After the change, no exception is thrown anymore. I think this is not the desired behavior.

Reply:
Alright, then:

```
        if (flywayContainer.isValidateAtStart()) {
            if (flywayContainer.isCleanOnValidationError()) {
                var result = flywayContainer.getFlyway().validateWithResult();
                
                if (!result.validationSuccessful) {
                    flywayContainer.getFlyway().clean();
                }
            }
            else {
                flywayContainer.getFlyway().validate();
            }
        }
```

Reply:
Done in ca5069d095464c055245862bbe5c76630bb4d3c2